doc/sphinx/language/core/records.rst

```diff
@@ -136,13 +139,55 @@ synonym :cmd:`Structure`.
       .. coqtop:: all
 
          Record MyRecord := { myfield : nat } as VarName.
-         About myfield. (* observe the MyRecord variable is named VarName *)
+         About myfield. (* observe the MyRecord variable is named "VarName" *)
          
-         (* This allows us to make VarName implicit for instance, without having to rename the variable as would be the case without the as clause. *)
+         (* make "VarName" implicit without having to rename the variable,
+            which would be necessary without the "as" clause *)
          Arguments myfield {VarName}.   (* make "VarName" an implicit parameter *)
          Check myfield.
          Check (myfield (VarName:={| myfield := 0 |})).
 
+   .. exn:: Records declared with the keyword Record or Structure cannot be recursive.
+
+      The record name :token:`ident` appears in the type of its fields, but uses
+      the :cmd:`Record` command. Use  the :cmd:`Inductive` or
+      :cmd:`CoInductive` command instead.
+
+   .. exn:: Cannot handle mutually (co)inductive records.
+
+      Records cannot be defined as part of mutually inductive (or
+      coinductive) definitions, whether with records only or mixed with
+      standard definitions.
+
+   .. exn:: @ident already exists
+
+      The fieldname :n:`@ident` is already defined as a global.
+
+   .. warn:: @ident cannot be defined.
+
+     It can happen that the definition of a projection is impossible.
+     This message is followed by an explanation of this impossibility.
+     There may be two reasons:
+
+     #. The :term:`body` of :token:`ident` uses an incorrect elimination for
+        :token:`ident` (see :cmd:`Fixpoint` and :ref:`Destructors`).
+     #. The type of the projections :token:`ident` depends on previous
+        projections which themselves could not be defined.
+
+   .. warn:: @ident__field cannot be defined because it is informative and @ident__record is not
```

Comment:
I wonder whether it is really appropriate to use `field` to characterize the first ident: if I'm not mistaken, the field is correctly defined, it's just the projection that is not.

Reply:
The grammar and text refers to the field identifier, not the projection identifier. 
 Shortly below, we appropriately refer to it as `ident__field`.  Not perfect but the best that can be done.  I added some words.